### PR TITLE
Do not allow KeyCombinations from modifier keys

### LINKED ISSUE #3869: Setting shift as the back button disallows typing
As title, if you set the shift key as the back button, you cannot type in any text boxes. Instead, you exit to the previous screen.

osu!lazer version: 2018.1214.0

### PR DESCRIPTION
**Pull request by UselessToucan**
Closes #3869

### CODE REVIEW

File: osu.Game/Overlays/KeyBinding/KeyBindingRow.cs [hunk 1]
```diff
@@ -216,8 +216,11 @@ protected override bool OnKeyDown(KeyDownEvent e)
                 }
             }
 
-            bindTarget.UpdateKeyCombination(KeyCombination.FromInputState(e.CurrentState));
-            if (!isModifier(e.Key)) finalise();
+            if (!isModifier(e.Key))
+            {
+                bindTarget.UpdateKeyCombination(KeyCombination.FromInputState(e.CurrentState));
```
Comment (peppy) [inline osu.Game/Overlays/KeyBinding/KeyBindingRow.cs]: i think this line should still be outside so the UI updates?

Comment (ringosham) [reply osu.Game/Overlays/KeyBinding/KeyBindingRow.cs]: I use the shift key to pause the game in stable. So would it be best to just do nothing when the user is in a textbox?

**Comment by peppy** [comment]
We can't actually have this as a thing. There's a reason modifier keys are allowed to be bound (see things like osu!catch / osu!mania, which use shift as a valid and default gameplay key).

**Comment by peppy** [comment]
Closing as not a valid fix. Will need some re-thinking.

**Pull request closed without merging**